packages/components/src/form-file-upload/index.tsx

```diff
@@ -50,9 +50,16 @@ export function FormFileUpload( {
 	// @todo: Temporary fix a bug that prevents Chromium browsers from selecting ".heic" files
 	// from the file upload. See https://core.trac.wordpress.org/ticket/62268#comment:4.
 	// This can be removed once the Chromium fix is in the stable channel.
-	const compatAccept = !! accept?.includes( 'image/*' )
-		? `${ accept }, image/heic, image/heif`
-		: accept;
+	const [ isSafari, setIsSafari ] = useState( false );
+	useEffect( () => {
+		setIsSafari(
+			/^((?!chrome|android).)*safari/i.test( navigator.userAgent )
+		);
+	}, [] );
```

Comment:
If I do a similar thing to get Safari like in `duotone.js`.
```
const isSafari =
	window?.navigator.userAgent &&
	window.navigator.userAgent.includes( 'Safari' ) &&
	! window.navigator.userAgent.includes( 'Chrome' ) &&
	! window.navigator.userAgent.includes( 'Chromium' );
```
eslint alerts: `Use of DOM global 'window' is forbidden in module scope`.

Also, checking the `userAgent` with a variable will trigger:
`Use of DOM global 'navigator' is forbidden in the render-cycle of a React FC, consider moving this inside useEffect()`.


I'm not really convinced this is a good fix. 
Pinging @WordPress/gutenberg-components for a quality check 😅

Reply:
This seems to me like a reasonable and unavoidable bandaid fix in this case 👍

Reply:
May be wrong but moving the `const isSafari = ...` to the top of the file (after the imports) won't trigger the error? If not, there seems to be a function `isSafary()` in the build, imported from @ariakit?

Reply:
The point of executing that in a `useEffect` is so we don't break SSR use cases.

> there seems to be a function isSafari() in the build, imported from https://github.com/ariakit?

Good find! But that isn't really a public API and the underlying detection mechanisms seem [similar](https://github.com/ariakit/ariakit/blob/d955bf36167ff0d9175dd69caac39ae30c9962d3/packages/ariakit-core/src/utils/platform.ts#L21) anyway. We can always encapsulate it ourselves for reuse if we need this kind of browser detection more.

Reply:
Right, browser detection from userAgent is generally not recommended. This seems to be the second place it's needed though, after duotone.js. This may be pretty short-lived considering one of the causes for it is already fixed in Chromium and is in Chrome Canary. Seems WP 6.7.1 may end up with reverting this fix completely.

Reply:
> The point of executing that in a useEffect is so we don't break SSR use cases.

Seems like a weird hack for SSR :( Btw, using lazy initialization for state silences the warnings.

```js
const [ isSafari ] = useState( () =>
	/^((?!chrome|android).)*safari/i.test( navigator?.userAgent )
);
```

Reply:
The goal here is to reference `window` or `navigator` in a safe way, so that it doesn't crash when run in Node.js which doesn't have these globals.

The `ssr-friendly` ESLint plugin is not perfect and it both accepts code that is bad and rejects code that is good.

This is OK but `ssr-friendly` will complain:
```js
const isSafari = typeof window !== 'undefined' ? window.navigator.userAgent.includes(...) : false;
```
And this @mamaduka's suggestion will crash because the `useState` initializer is called during SSR. `ssr-friendly` just doesn't detect it:
```js
const [ isSafari ] = useState( () => window.navigator.userAgent.includes(...) );
```

Be careful that this will also crash in Node.js despite the optional chaining:
```js
window?.navigator
```
Because the `window` global doesn't exist, the code will crash even before getting to apply the `?.` operator. That's why the `typeof window` check is done, because `typeof` can deal with undefined variables.

One of good solutions is also using `globalThis`:
```js
globalThis.window?.navigator.userAgent
globalThis.navigator?.userAgent
```
The `globalThis` variable is always present, and in browsers it has the `window` and `navigator` fields, in Node.js it doesn't have them. The `?.` operator works here because we're now dealing with fields of an object, not with global variables.

I think we shouldn't need `useEffect` here. The UA check can be done in one pass, the component doesn't need to be rendered twice.

So, all you need is to find a solution that:
1. Actually works and doesn't crash, even if `ssr-friendly` doesn't warn you
2. Satisfies the `ssr-friendly` rules. You might need to reshuffle code that is already good, e.g., by wrapping it in a function etc.

Reply:
Thanks @jsnajdr !

I updated with the `globalThis` approach in https://github.com/WordPress/gutenberg/pull/67139/commits/21a1cb05850ca2c052f59a659fc339848aa9a27b

Can I get a green check?